### PR TITLE
Ac/sensors

### PR DESCRIPTION
**Pull request by kdpatino**
* UV Kernel driver

### CODE REVIEW

File: matrixio-core.c [hunk 1]
```diff
@@ -77,7 +77,8 @@ int matrixio_hw_reg_read(void *context, unsigned int reg, unsigned int *val)
 	hw_addr->burst = 0;
 	hw_addr->readnwrite = 1;
 
-	ret = matrixio_spi_transfer(matrixio->spi, send_buf, (uint8_t *)recv_buf, 4);
+	ret = matrixio_spi_transfer(matrixio->spi, send_buf,
+				    (uint8_t *)recv_buf, 4);
```
Comment (arhuaco) [inline matrixio-core.c]: Wondering... Why send buf 4 and recv buf 2? Is the address in send buff? (learning)

File: matrixio-core.c [hunk 2]
```diff
@@ -77,7 +77,8 @@ int matrixio_hw_reg_read(void *context, unsigned int reg, unsigned int *val)
...
 
 	if (ret < 0) {
 		*val = 0;
```
Comment (arhuaco) [inline matrixio-core.c]: No need to modify val here. The return value should be enough.

File: matrixio-core.c [hunk 3]
```diff
@@ -77,7 +77,8 @@ int matrixio_hw_reg_read(void *context, unsigned int reg, unsigned int *val)
 	hw_addr->burst = 0;
 	hw_addr->readnwrite = 1;
```
Comment (arhuaco) [inline matrixio-core.c]: mutex_lock(&matrixio_spi_lock); and unlock somewhere in this function?

File: matrixio-core.c (hunk 2 above)
Comment (arhuaco) [reply matrixio-core.c]: Below I see *val = recv_buf[1];
Doesn't this cause a memory overflow?

File: matrixio-core.c [hunk 4]
```diff
@@ -103,36 +104,68 @@ int matrixio_hw_reg_write(void *context, unsigned int reg, unsigned int val)
 
 	send_buf[1] = val;
```
Comment (arhuaco) [inline matrixio-core.c]: Isn't there a memory overflow in this line?

send_buf[1] = val;

Also, why isn't there a mutex in this function?

File: matrixio-core.c [hunk 5]
```diff
@@ -103,36 +104,68 @@ int matrixio_hw_reg_write(void *context, unsigned int reg, unsigned int val)
...
-		 .name = "matrixio-uart",
-		 .of_compatible = "matrixio-uart",
- 		 .platform_data = matrixio,
-		 .pdata_size = sizeof(*matrixio),
-		}
+	int ret;
+	int offset;
+	unsigned int val;
+	uint16_t *words = data;
+
```
Comment (arhuaco) [inline matrixio-core.c]: mutex?

File: matrixio-core.c [hunk 6]
```diff
@@ -150,15 +183,16 @@ static int matrixio_init(struct matrixio *matrixio,
 		return ret;
 	}
 
-	if(pdata && pdata->platform_init) {
+	if (pdata && pdata->platform_init) {
 		ret = pdata->platform_init(matrixio->dev);
-		if(ret != 0) {
+		if (ret != 0) {
```
Comment (arhuaco) [inline matrixio-core.c]: You've been doing "if(ret)" in the code. It's better to be consistent. It's OK like this though.

File: matrixio-core.c (hunk 1 above)
Comment (eljuguetero) [reply matrixio-core.c]: Due to SPI intrinsics..  when one bit is written, one bit is read (in the same clock event) .  The internal FSM in the FPGA don't have noting to write when the first 2 bytes was written.

File: matrixio-core.c (hunk 2 above)
Comment (eljuguetero) [reply matrixio-core.c]: it does not cause overflow, (*val), is a value of 32 bits, recv_buf[1] is a value of 16 bits.

File: matrixio-core.c (hunk 3 above)
Comment (eljuguetero) [reply matrixio-core.c]: mmmm, I have the same doubt...  it is probably better define a mutex section.

File: matrixio-core.c (hunk 2 above)
Comment (arhuaco) [reply matrixio-core.c]: Yeah. I misread this.

File: matrixio-core.c (hunk 3 above)
Comment (arhuaco) [reply matrixio-core.c]: Maybe in other functions as well.

**Pull request merged**